src/vec.rs

```diff
@@ -120,18 +124,44 @@ named!(range_literal <CompleteByteSlice, usize>, do_parse!(
 	(num * suffix)
 ));
 
+type ResolvedRange = (usize, Option<usize>);
+
+fn delimited_range(input: CompleteByteSlice) -> IResult<CompleteByteSlice, Option<ResolvedRange>> {
+    ws!(input,
+        opt!(
+            delimited!(
+                char!('['),
+                do_parse!(
+                    range: range_literal >>
+                    opt!(tag!(":")) >>
```

Comment:
This `opt!` would allow queries like `aggregate(…) [5m]` when proper subqueries support will be implemented, and that, AFAICT, it still prohibited¹. It might be useful as an optional extension, but for now ranges should have separate parsers or, better yet, this parser should accept a `bool` argument that would allow or deny regular range selectors like `[5m]`.

¹ [Error executing query: invalid parameter 'query': 1:23: parse error: ranges only allowed for vector selectors](http://demo.robustperception.io:9090/graph?g0.range_input=1h&g0.expr=count(node_load1%20%3E%201)%20%5B5m%5D&g0.tab=1)

Reply:
I tried adding a boolean argument but I felt that made the usage more confusing. Instead I added a separate subquery parser and I modified the top-level vector parser to allow either alternate. Please let me know what you think.